src/utils/src/lib.rs

```diff
@@ -18,3 +23,30 @@ macro_rules! debug {
         ()
     };
 }
+
+// -----------------------------------------------------------------------------
+/// Enum containing error types.
+pub enum Error {
+    Io(io::Error),
+    Path,
+}
+
+impl std::fmt::Display for Error {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "Erorr: {}", (*self))
+    }
+}
+
+/// A safe wrapper around libc::mkfifo().
+pub fn mkfifo<P: AsRef<Path>>(fifo_path: P, mode: u32) -> result::Result<(), Error> {
+    let fifo_path_str = fifo_path.as_ref().to_str().ok_or(Error::Path)?;
+    let fifo_path_cstr = CString::new(fifo_path_str).unwrap();
+
+    let ret = unsafe { libc::mkfifo(fifo_path_cstr.as_ptr(), mode) };
```

Comment:
Unsafe code blocks should be accompanied by comments saying why they're actually safe. In this case we can say that we checked that `fifo_path_cstr` is valid, and we checked the return of the unsafe code.